dataactbroker/handlers/accountHandler.py

```diff
@@ -239,24 +243,42 @@ def email_users(self, system_email):
 def perms_to_affiliations(perms):
     """Convert a list of perms from MAX to a list of UserAffiliations. Filter
     out and log any malformed perms"""
-    available_codes = {
+    available_cgacs = {
         cgac.cgac_code: cgac
         for cgac in GlobalDB.db().session.query(CGAC)
     }
+    available_frecs = {
+        frec.frec_code: frec
+        for frec in GlobalDB.db().session.query(FREC)
+    }
     for perm in perms:
         components = perm.split('-PERM_')
         if len(components) != 2:
             logger.warning('Malformed permission: %s', perm)
             continue
 
-        cgac_code, perm_level = components
+        codes, perm_level = components
         perm_level = perm_level.lower()
-        if cgac_code not in available_codes or perm_level not in 'rws':
+        split_codes = codes.split('-FREC_')
+        frec_code = cgac_code = None
+        if len(split_codes) == 2:
```

Comment:
It would be nice to have some comments here so we know what's going on and why we're checking these lengths.